demisto_sdk/commands/common/hook_validations/content_entity_validator.py

```diff
@@ -250,6 +254,52 @@ def should_run_fromversion_validation(self):
 
         return True
 
+    @error_codes('BA117')
+    def are_fromversion_and_toversion_in_correct_format(self) -> bool:
+
+        if self.file_path.endswith('.json'):
+            from_version = self.current_file.get('fromVersion', '00.00.00') or '00.00.00'
+            to_version = self.current_file.get('toVersion', '00.00.00') or '00.00.00'
+        elif self.file_path.endswith('.yml'):
+            from_version = self.current_file.get('fromversion', '00.00.00') or '00.00.00'
+            to_version = self.current_file.get('toversion', '00.00.00') or '00.00.00'
+        else:
+            return True
+
+        if not FROM_TO_VERSION_REGEX.fullmatch(from_version):
+            error_message, error_code = Errors.from_and_to_version_are_incorrect_format(
+                'fromversion')
+            self.handle_error(error_message, error_code, file_path=self.file_path)
+            return False
+        if not FROM_TO_VERSION_REGEX.fullmatch(to_version):
+            error_message, error_code = Errors.from_and_to_version_are_incorrect_format(
+                'tovesion')
+            self.handle_error(error_message, error_code, file_path=self.file_path)
+            return False
+
+        return True
+
+    @error_codes('BA118')
+    def are_fromversion_toversion_synchronized(self) -> bool:
+
+        if self.file_path.endswith('.json'):
+            from_version = self.current_file.get('fromVersion', '')
+            to_version = self.current_file.get('toVersion', '')
+        elif self.file_path.endswith('.yml'):
+            from_version = self.current_file.get('fromversion', '')
+            to_version = self.current_file.get('toversion', '')
+        else:
+            return True
+
+        if not from_version or not to_version:
+            return True
```

Comment:
please add debug logging here too: `either not from_version or not to_version, considering them synced`